includes/tracking-events.php

```diff
@@ -0,0 +1,120 @@
+<?php
+    if(class_exists('WooCommerce')){
+        // /** check certain tracking scenarios **/
+        // if visitor is viewing product
+        if(!$this->single_item_tracked && is_product()){
+            $product = $this->resolve_product(get_queried_object_id());
+            $product_id = method_exists($product, 'get_id') ? $product->get_id() : $product->id;
+            $this->put_event_in_queue(
+                "window.metrilo.viewProduct('"
+                . $product_id
+                . "');"
+            );
+            $this->single_item_tracked = true;
+        }
+        
+        // if visitor is viewing product category
+        if (!$this->single_item_tracked && is_product_category()) {
+            $category = get_queried_object();
+            $this->put_event_in_queue(
+                "window.metrilo.viewCategory('"
+                . $category->term_id
+                . "');"
+                
+            );
+            $this->single_item_tracked = true;
+        }
+    
+        // if visitor is viewing search page
+        if (!$this->single_item_tracked && is_search()) {
+            $search = get_queried_object();
+            $this->put_event_in_queue(
+                "window.metrilo.search('"
+                . get_search_query()
+                . "', '"
+                . get_search_link()
+                . "');"
+        
+            );
+            $this->single_item_tracked = true;
+        }
+        
+        // if visitor is viewing shopping cart page
+        if(!$this->single_item_tracked && is_cart()){
+            $this->put_event_in_queue(
+                "window.metrilo.customEvent('view_cart');"
+            );
+            $this->single_item_tracked = true;
+        }
+        // if visitor is anywhere in the checkout process
+        if (!$this->single_item_tracked && is_order_received_page()) {
```

Comment:
You can combine this block and two others at lines 80 and 89. They do the same thing.

Reply:
Fixed

Reply:
Not fixed. Resolving this because I added new comments on the new version of the code.